src/JuMP.jl

```diff
@@ -107,6 +145,12 @@ mutable struct Model <: AbstractModel
 
     customnames::Vector
 
+    # Factory used to create a new optimizer, it is kept as it might be needed
+    # again if the user requests a copy of the model using `Base.copy`.
+    # In Manual and Automatic mode: Factory used to create the optimizer or
+    #                               Nothing if it has not already been set
+    # In Direct mode: Nothing
+    factory::Union{Nothing, Factory}
     # In Manual and Automatic modes, LazyBridgeOptimizer{CachingOptimizer}.
     # In Direct mode, will hold an AbstractOptimizer.
     moibackend::MOI.AbstractOptimizer
```

Comment:
Could this be an `MOI.ModelLike` to support non-optimizer backends?

Reply:
Yes, I don't remember which PR removed that possibility. The model of the CachingOptimizer can be a `MOI.ModelLike` and `CachingOptimizer` will still be an `AbstractOptimizer` so this just removes the possibility to use a non-optimizer in direct mode. Do we want to allow non-optimizer in direct mode ?